idaes/core/solvers.py

```diff
@@ -0,0 +1,39 @@
+from pyomo.environ import SolverFactory
+import idaes
+
+class SolverWrapper(object):
+    def __init__(self, name):
+        self.name = name
+        if name == 'default':
+            self.solver = None
+            doc = "IDAES Configured Default Solver"
+        else:
+            self.solver = SolverFactory.get_class(name)
+            doc = SolverFactory.doc(name)
+        SolverFactory.unregister(name)
+        # Re-register the solver (register is a decorator)
+        SolverFactory.register(name, doc)(self)
```

Comment:
Should there be a test to make sure we don't override a solver twice?  Something like:
```python
try:
    if name in SolverFactory:
        self.solver = SolverFactory.get_class(name)
        doc = SolverFactory.doc(name)
    else:
        self.solver = None
        if name == 'default':
            doc = "IDAES Configured Default Solver"
        else
            doc = ""
    if type(self.solver) is not SolverWrapper:
        SolverFactory.unregister(name)
        # Re-register the solver (register is a decorator)
        SolverFactory.register(name, doc)(self)

Reply:
I'm not sure.  Since this happens directly when you import a module isn't it already protected from happening twice?

Reply:
Yes - although if you make a function that adds the wrappers (that the user calls), then there is a distinct possibility that the wrappers could be created twice.